src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] internalArray;
+    private int size;
+
+    public ArrayList() {
+        this.internalArray = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        this.internalArray = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size != internalArray.length) {
+            for (int i = 0; i < internalArray.length; i++) {
+                if (i == size) {
+                    internalArray[i] = value;
+                    size++;
+                    break;
+                }
+            }
+        } else {
+            grow();
+            internalArray[size] = value;
+            size++;
+        }
```

Comment:
Don't complicate if-else construction. The for-loop inside the `if` block is unnecessary since you are only assigning a value at the `size` index. You can simplify this by directly assigning the value without the loop.